irispy/iris_tools.py

```diff
@@ -802,3 +803,25 @@ def _apply_or_undo_exposure_time_correction(sequence, correction_function):
         else:
             converted_data_list.append(cube)
     return converted_data_list
+
+def calculate_dust_mask(data_array):
+    """Calculate a mask with the dust positions in a given arrayself.
+
+    Parameters
+    ----------
+    data_array : `numpy.ndarray`
+        This array contains some dust poisition that will be calculated. The array
+        must have scaled values.
+
+    Returns
+    -------
+    dust : `numpy.ndarray` of `bool`
+        This array has the same shape than data_array and contains the dust positions
+        when the value is True.
+
+    """
+    dust = (data_array < 0.5) & (data_array > -200)
+    struct = ndimage.generate_binary_structure(2, 2)
+    for i in range(data_array.shape[0]):
+        dust[i] = ndimage.binary_dilation(dust[i], structure=struct).astype(dust.dtype)
```

Comment:
I think this can be simplified to the following:
```python
mask_array[(data_array < 0.5) & (data_array > -200)] = True
return mask
```

Reply:
I am not agree because a dust pixel can affect his neighbours so I have to use `dilate` to remove these pixels.

Reply:
The problem of your code line is that mask_array is now a 1D array of boolean, that is why I used something else.

Reply:
You should not need a loop here. You can use `binary_dilation` with any array shape, just make sure that `struct` is appropriate (e.g. zero along the dimensions you don't want to dilate).

Reply:
Ah OK.  I didn't get the significance of the ```dilate```.  However, just for reference, my example shouldn't result in a 1D array.  The shape should be maintained.  That could be useful for you in the future.

Reply:
Oh yes, I don't know why I think to that thanks Tiago.

Reply:
Yes Dan, I was not using the first lines you wrote so that was not working. I modified it so.